tamr_client/dataset/record.py

```diff
@@ -143,3 +143,17 @@ def _delete_command(record: Dict, *, primary_key_name: str) -> Dict:
         The DELETE command in the proper format
     """
     return {"action": "DELETE", "recordId": record[primary_key_name]}
+
+
+def stream(session: Session, dataset: Dataset) -> Generator[Dict, None, None]:
+    """Stream the records in this dataset as Python dictionaries.
+
+    Args:
+        dataset: Dataset from which to stream records
+
+    Returns:
+        Python generator yielding records
+    """
+    with session.get(str(dataset.url) + "/records", stream=True) as response:
```

Comment:
You should use [tc.response.ndjson](https://github.com/Datatamer/tamr-client/blob/c08704de48eacb0d8f992ca9dfca80df17018d5b/tamr_client/response.py#L34) for this instead.